.github/actions/javascript/proposalPoliceCommentEdit/proposalPoliceCommentEdit.ts

```diff
@@ -0,0 +1,114 @@
+import {context, getOctokit} from '@actions/github';
+import InitOpenAI from 'openai';
+import CONST from '@github/libs/CONST';
+import type {GitHubType} from '@github/libs/GithubUtils';
+
+const OpenAI = new InitOpenAI({apiKey: process.env.OPENAI_API_KEY});
+
+async function processIssueCommentEdit(octokit: InstanceType<typeof GitHubType>) {
+    const payload = context.payload;
+    const OPENAI_ASSISTANT_ID = process.env.OPENAI_ASSISTANT_ID;
+
+    // check if the issue is open and the has labels
+    if (payload.issue?.state !== 'open' && !payload.issue?.labels.some((issueLabel: {name: string}) => issueLabel.name === CONST.LABELS.HELP_WANTED)) {
+        return;
+    }
+
+    if (!OPENAI_ASSISTANT_ID) {
+        console.log('OPENAI_ASSISTANT_ID missing from the environment variables');
+        return;
+    }
+
+    // You need to adapt this part to fit the Edit Use Case as in the original function
+    const content = `I NEED HELP WITH CASE (2.) WHEN A USER THAT POSTED AN INITIAL PROPOSAL OR COMMENT (UNEDITED) THEN EDITS THE COMMENT - WE NEED TO CLASSIFY THE COMMENT BASED IN THE GIVEN INSTRUCTIONS AND IF TEMPLATE IS FOLLOWED AS PER INSTRUCTIONS. IT IS MANDATORY THAT YOU RESPOND ONLY WITH "NO_ACTION" IN CASE THE COMMENT IS NOT A PROPOSAL. \n\nPrevious comment content: ${payload.changes.body?.from}.\n\nEdited comment content: ${payload.comment?.body}`;
+
+    // create thread with first user message and run it
+    const createAndRunResponse = await OpenAI.beta.threads.createAndRun({
+        /* eslint-disable @typescript-eslint/naming-convention */
+        assistant_id: OPENAI_ASSISTANT_ID ?? '',
+        thread: {messages: [{role: 'user', content}]},
+    });
+
+    // count calls for debug purposes
+    let count = 0;
+    // poll for run completion
+    const intervalID = setInterval(() => {
```

Comment:
Can we wrap this `setInterval` polling in some async function such that we can just do:

```ts
const {response} = await OpenAIUtils.prompt(prompt);
```

Reply:
Done, created two `OpenAIUtils` functions: `prompt` and `promptEdit` since the pooling functions are slightly different between `created` and `edited` comments.